### ISSUE TITLE
Deploy from travis

### ISSUE THREAD

**Issue by david4096** [ga4gh/server#1569]
It is possible to automatically release from successful travis builds using https://docs.travis-ci.com/user/deployment/pypi/ .

**Comment by dcolligan** [comment]
This came up a while back but we elected not to do it; I forget why.  What do you see as the advantages / disadvantages of this?

**Comment by david4096** [comment]
Basically, it's less time thinking about how to do a release and more time writing code. Having successful master merges release from Travis is a nice way to do continuous release.

Having continuous/nightly builds instead of manual alpha builds would let us test out the PyPi machinery between tagged releases without any work. If Travis never overwrites the tagged releases, it can be a good source for alpha builds without any real risk of harm (other than filling pypi with a lot of quickly obsoleted releases).

**Comment by dcolligan** [comment]
It could be worth investigating, but the only step this really saves us is uploading the source tarball to pypi, which is one less step of N, I suppose.

**Comment by dcolligan** [comment]
I'm not sold on the nightly alpha builds idea, but I do think auto-releases from Travis is a good idea.  The only drawback, so far as I can see, is that the release to pypi will not get done for up to 24 hours, since the push will be triggered by a cron job, not a PR build (since the release is tagged only after the release commit is merged into master).  If we have a requirement of pushing to pypi immediately after a release, then we shouldn't go this route, but so far as I know that's not a concern.  (In the event of an emergency, we can always fall back on manual uploading steps.)

**Comment by dcolligan** [comment]
Er, that drawback can be mitigated as well since Travis builds can be manually triggered http://stackoverflow.com/questions/17606874/trigger-a-travis-ci-rebuild-without-pushing-a-commit

**Comment by kozbo** [comment]
Is it a common pattern to store all of our intermediate builds on Pypi? Will this confuse our Python users as to which version to download?

**Comment by dcolligan** [comment]
Automatic deployments from ga4gh-common are working as of this release: https://github.com/ga4gh/ga4gh-common/releases/tag/v0.0.6

**Comment by dcolligan** [comment]
Just for future reference, the command I found that worked was:

`$ travis setup pypi --repo ga4gh/ga4gh-common --debug --force` (modulo the appropriate repo name)

The `travis encrypt` command that they recommend was a bit funky.

**Comment by david4096** [comment]
Very cool. So you tag a release and then cron will put it in PyPi for you?

**Comment by dcolligan** [comment]
Builds are also triggered by new tags, I found, so the push to pypi happens almost immediately.  Technically, we always announce a release before it hits pypi (because tagging a release on github sends out an email to all the repo's subscribers).  We should keep the existing docs, since there may come a time when we need to do a manual release, but we should add a section explaining how we it expect it to be usually done, the automated way.  I'll do that.

**Comment by dcolligan** [comment]
Code committed in #1576

**Comment by dcolligan** [comment]
Confirmed to work https://travis-ci.org/ga4gh/server/builds/205958552

**Comment by dcolligan** [comment]
schemas PR here: https://github.com/ga4gh/schemas/pull/825

client PR here: https://github.com/ga4gh/ga4gh-client/pull/66
